libafl_qemu/src/emu/hooks.rs

```diff
@@ -1118,24 +1118,25 @@ where
         modules
     }
 
-    pub fn first_exec_all(&mut self) {
+    pub fn first_exec_all(&mut self, state: &mut S) {
         unsafe {
             self.modules_mut()
-                .first_exec_all(Self::emulator_modules_mut_unchecked());
+                .first_exec_all(Self::emulator_modules_mut_unchecked(), state);
         }
     }
 
-    pub fn pre_exec_all(&mut self, input: &S::Input) {
+    pub fn pre_exec_all(&mut self, input: &S::Input, state: &mut S) {
```

Comment:
I think the state should be the first parameter to follow the other parts of LibAFL